### PR TITLE
Compact serialization codecov test results issues[API-1791]

### PR DESCRIPTION
**Pull request by harunalpak**
Some unused methods removed, typo fixes.

And 2 test were written to check throwing error when field is undefined or field type not matched.

### CODE REVIEW

File: src/serialization/compact/SchemaService.ts [hunk 1]
```diff
@@ -111,11 +111,6 @@ export class SchemaService {
             this.schemas.set(schemaId.toString(), schema);
             return;
         }
-        if (!existingSchema.equals(schema)) {
-            throw new IllegalStateError(
-                `Schema with schemaId ${schemaId} already exists. existing schema: ${existingSchema} new schema: ${schema}`
-            );
-        }
```
Comment (srknzl) [inline src/serialization/compact/SchemaService.ts:118]: Why did you delete this?

File: src/serialization/compact/Schema.ts [hunk 2]
```diff
@@ -101,33 +101,4 @@ export class Schema {
...
-    equals(other: Schema): boolean {
-        if (this === other) {
-            return true;
-        }
-        return this.numberVarSizeFields === other.numberVarSizeFields &&
-            this.fixedSizeFieldsLength === other.fixedSizeFieldsLength &&
-            this.typeName === other.typeName &&
-            this.schemaId.equals(other.schemaId) &&
-            this.hasSameFields(other);
-    }
```
Comment (srknzl) [inline src/serialization/compact/Schema.ts:132]: These should be stay I think

**Review by srknzl — COMMENTED** [review]
Except the schema.equals, looks good

File: src/serialization/compact/SchemaService.ts (hunk 1 above)
Comment (harunalpak) [reply src/serialization/compact/SchemaService.ts:118]: I confused with other control inside put method, I supposed that this was checked inside "put" method. I get it back.

File: src/serialization/compact/Schema.ts (hunk 2 above)
Comment (harunalpak) [reply src/serialization/compact/Schema.ts]: Fixed

**Review by srknzl — APPROVED** [review]
